aiohttp/helpers.py

```diff
@@ -1076,3 +1081,13 @@ def status_code_must_be_empty_body(code: int) -> bool:
     """Check if a status code must return an empty body."""
     # https://datatracker.ietf.org/doc/html/rfc9112#section-6.3-2.1
     return code in {204, 304} or 100 <= code < 200
+
+
+def should_remove_content_length(method: str, code: int) -> bool:
+    """Check if a Content-Length header should be removed.
+
+    This should always be a subset of must_be_empty_body
+    """
+    # https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6-7
+    # https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6-8
+    return code in (204, 304) or 100 <= code < 200 or method == hdrs.METH_CONNECT
```

Comment:
For CONNECT requests, it only applies if the response is 2xx, so I think there's a small mistake here.

Also, the wording for HEAD and 304 are basically identical, so I'm not 100% sure we should include 1 here and not the other. I think we really just need to ensure that any automatic Content-Length header is excluded in both cases.

Reply:
Just realised the 2xx CONNECT is also an issue with the previously merged functions.

Reply:
I will prepare another PR to fix that

Reply:
https://github.com/aio-libs/aiohttp/pull/7772

I will adjust the logic in this PR shortly

Reply:
logic adjusted in 5630ca593341df97d5551d3dd175163afcbc05d6

Reply:
Tests adjusted in 79d55e28b248c48b90b53eef0d0239fdc1d17728